File: sdk-swift/Sources/DecibelSDK/Models/Position.swift
import Foundation

public struct UserPosition: Codable {
    public let market: String
    public let user: String
    public let size: Double
    public let userLeverage: Double
    public let entryPrice: Double
    public let isIsolated: Bool
    public let unrealizedFunding: Double
    public let estimatedLiquidationPrice: Double
    public let tpOrderID: String?
    public let tpTriggerPrice: Double?
    public let tpLimitPrice: Double?
    public let slOrderID: String?
    public let slTriggerPrice: Double?
    public let slLimitPrice: Double?
    public let hasFixedSizedTpsls: Bool

    enum CodingKeys: String, CodingKey {
        case market, user, size
        case userLeverage = "user_leverage"
        case entryPrice = "entry_price"
        case isIsolated = "is_isolated"
        case unrealizedFunding = "unrealized_funding"
        case estimatedLiquidationPrice = "estimated_liquidation_price"
        case tpOrderID = "tp_order_id"
        case tpTriggerPrice = "tp_trigger_price"
        case tpLimitPrice = "tp_limit_price"
        case slOrderID = "sl_order_id"
        case slTriggerPrice = "sl_trigger_price"
        case slLimitPrice = "sl_limit_price"
        case hasFixedSizedTpsls = "has_fixed_sized_tpsls"
    }
}

File: sdk-rust/src/models/position.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UserPosition {
    pub market: String,
    pub user: String,
    pub size: f64,
    pub user_leverage: f64,
    pub entry_price: f64,
    pub is_isolated: bool,
    pub unrealized_funding: f64,
    pub estimated_liquidation_price: f64,
    pub tp_order_id: Option<String>,
    pub tp_trigger_price: Option<f64>,
    pub tp_limit_price: Option<f64>,
    pub sl_order_id: Option<String>,
    pub sl_trigger_price: Option<f64>,
    pub sl_limit_price: Option<f64>,
    pub has_fixed_sized_tpsls: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct PerpPosition {
    pub size: f64,
    pub sz_decimals: i32,
    pub entry_px: f64,
    pub max_leverage: f64,
    pub is_long: bool,
    pub token_type: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CrossedPosition {
    pub positions: Vec<PerpPosition>,
}
